src/AspNet.Security.OAuth.Amazon/AmazonAuthenticationOptions.cs

```diff
@@ -20,28 +21,30 @@ public class AmazonAuthenticationOptions : OAuthOptions
         /// </summary>
         public AmazonAuthenticationOptions()
         {
-            AuthenticationScheme = AmazonAuthenticationDefaults.AuthenticationScheme;
-            DisplayName = AmazonAuthenticationDefaults.DisplayName;
             ClaimsIssuer = AmazonAuthenticationDefaults.Issuer;
 
-            CallbackPath = new PathString(AmazonAuthenticationDefaults.CallbackPath);
+            CallbackPath = AmazonAuthenticationDefaults.CallbackPath;
 
             AuthorizationEndpoint = AmazonAuthenticationDefaults.AuthorizationEndpoint;
             TokenEndpoint = AmazonAuthenticationDefaults.TokenEndpoint;
             UserInformationEndpoint = AmazonAuthenticationDefaults.UserInformationEndpoint;
 
             Scope.Add("profile");
             Scope.Add("profile:user_id");
+
+            Fields.Add("email");
+            Fields.Add("name");
+            Fields.Add("user_id");
+
+            ClaimActions.MapJsonKey(ClaimTypes.Email, "email");
+            ClaimActions.MapJsonKey(ClaimTypes.Name, "name");
+            ClaimActions.MapJsonKey(ClaimTypes.NameIdentifier, "user_id");
+            ClaimActions.MapJsonKey(ClaimTypes.PostalCode, "postal_code");
         }
 
         /// <summary>
         /// Gets the list of fields to retrieve from the user information endpoint.
         /// </summary>
-        public ISet<string> Fields { get; } = new HashSet<string>
-        {
-            "email",
-            "name",
-            "user_id"
-        };
+        public ICollection<string> Fields { get; } = new HashSet<string>();
```

Comment:
Revert.

Reply:
Why? It's functionally equivalent because the fields are populated by the constructor now instead of inline and `ISet<T>` wasn't found when I tried to initially compile it for 2.0.

Reply:
Because it's not a necessary change and it makes the 2.0 signature inconsistent with what we have in the 1.x version. Only changes that are strictly necessary to make the providers work with 2.0 are accepted as part of the "2.0 port" PRs.